haystack/preview/components/file_converters/txt.py

```diff
@@ -0,0 +1,234 @@
+                        path,
+                        data.valid_languages,
+                    )
+
+            document = Document(content=text, meta=meta, id_hash_keys=data.id_hash_keys)
+            documents.append(document)
+
+        return self.output(documents=documents)
+
+    @staticmethod
+    def _prepare_metadata(meta: Optional[Union[Dict, List[Dict]]], file_paths: List[Union[str, Path]]) -> List[Dict]:
+        """
+        Prepare the metadata for the Documents.
+
+        :param meta: The metadata for the Documents.
+        :param file_paths: The paths to the text files.
+        """
+        if meta is None:
+            return [{"file_path": str(path)} for path in file_paths]
+
+        if isinstance(meta, dict):
+            meta = [meta] * len(file_paths)
+
+        if len(meta) != len(file_paths):
+            raise PipelineRuntimeError(
+                f"The number of meta entries must match the number of paths if meta is a list. "
+                f"Number of paths: {len(file_paths)}, number of meta entries: {len(meta)}"
+            )
+
+        return [{**m, "file_path": m.get("file_path", str(path))} for m, path in zip(meta, file_paths)]
+
+    @staticmethod
+    def _read_and_clean_file(path: Union[str, Path], encoding: str, remove_numeric_tables: bool) -> str:
+        """
+        Read and clean the text file.
+
+        :param path: The path to the text file.
+        :param encoding: The encoding of the text file.
+        :param remove_numeric_tables: Whether to remove numeric tables.
+        """
+        if not Path(path).exists():
+            raise PipelineRuntimeError(f"File at path {path} does not exist.")
+
+        with open(path, encoding=encoding) as file:
+            text = file.read()
+            pages = text.split("\f")
+            cleaned_pages = [TextfileToDocument._clean_page(page, remove_numeric_tables) for page in pages]
+            return "\f".join(cleaned_pages)
+
+    @staticmethod
+    def _clean_page(page: str, remove_numeric_tables: bool) -> str:
+        """
+        Clean a page of text from numeric tables if `remove_numeric_tables` is `True`.
+
+        :param page: The page of text.
+        :param remove_numeric_tables: Whether to remove numeric tables.
+        """
+        cleaned_lines = page.splitlines()
+        if remove_numeric_tables:
+            cleaned_lines = [line for line in cleaned_lines if not TextfileToDocument._is_numeric_row(line)]
+
+        return "\n".join(cleaned_lines)
+
+    @staticmethod
+    def _is_numeric_row(line: str) -> bool:
+        """
+        Check if a line is a numeric row. A line is considered a numeric row if it contains more than 40% digits and
+        does not end with a period.
+
+        :param line: The line to check.
+        """
+        words = line.split()
+        digits = [word for word in words if any(char.isdigit() for char in word)]
+        return len(digits) / len(words) > 0.4 and not line.strip().endswith(".")
```

Comment:
Is 0.4 some well-known ratio, perhaps it deserve a constant or more explanation, even parameter setting, idk?

Reply:
This was hard-coded in v1, I guess it makes sense to parametrize it.